spec/controllers/miq_report_controller/trees_spec.rb

```diff
@@ -157,8 +157,7 @@
       end
 
       def row_count_html(row_count_value)
-        "<label class='control-label col-md-2'>Row Count</label>\n" \
-        "<div class='col-md-10'>\n<p class='form-control-static'>#{row_count_value}</p>\n</div>"
+        {:cells => {:label => " Row Count", :value => row_count_value.to_s}}.to_json
```

Comment:
This method is called row_count_html, but it returns json?

Reply:
Earlier test cases were looking for an HTML content with the `row_count_value` capsuled within the html tags `<label>` and `<div>`

After converting to react, the test cases are now looking for a strings like:
`"{\"cells\":{\"label\":\" Row Count\",\"value\":\"7\"}}"`

from the variable (line#169 )`main_content` which has the data like:

`"<div id='flash_msg_div' style='display: none'>\n</div>\n\n<div id=\"react-vscvw1tj\"></div><script>\n//<![CDATA[\nManageIQ.component.componentFactory('MiqStructuredList', '#react-vscvw1tj', {\"title\":\"Basic Information\",\"headers\":null,\"rows\":[{\"cells\":{\"label\":\"Title\",\"value\":\"widget_0000000000002\"}},{\"cells\":{\"label\":\"Descrption\",\"value\":\"widget_0000000000002\"}},{\"cells\":{\"label\":\"Active\",\"value\":true}},{\"cells\":{\"label\":\"Default\",\"value\":false}}],\"message\":null,\"mode\":\"miq_summary miq_widget_basic_information\"})\n//]]>\n</script><div id=\"react-9fo675f5\"></div><script>\n//<![CDATA[\nManageIQ.component.componentFactory('MiqStructuredList', '#react-9fo675f5', {\"title\":\"Status\",\"headers\":null,\"rows\":[{\"cells\":{\"label\":\"Current Status\",\"value\":\"None\"}},{\"cells\":{\"label\":\"Last Run Time\",\"value\":\"\"}},{\"cells\":{\"label\":\"Message\",\"value\":\"Unknown\"}}],\"message\":null,\"mode\":\"miq_summary miq_widget_status\"})\n//]]>\n</script><div id=\"react-b7dd76ur\"></div><script>\n//<![CDATA[\nManageIQ.component.componentFactory('MiqStructuredList', '#react-b7dd76ur', {\"title\":\"Report Options\",\"headers\":null,\"rows\":[{\"cells\":{\"label\":\"Report Filter\",\"value\":\"*** Report no longer exists ***\"},\"style\":\"color_red\"},{\"cells\":{\"label\":\"Row Count\",\"value\":\"7\"}}],\"message\":null,\"mode\":\"miq_summary miq_widget_report_options\"})\n//]]>\n</script><div id=\"react-524on1gr\"></div><script>\n//<![CDATA[\nManageIQ.component.componentFactory('MiqStructuredList', '#react-524on1gr', {\"title\":\"Visibility\",\"headers\":null,\"rows\":[{\"cells\":{\"label\":\"Show\",\"value\":\"To All Users\"}}],\"message\":null,\"mode\":\"miq_summary miq_widget_visibility\"})\n//]]>\n</script>\n\n"`

Reply:
Oh I see, so it's json but embedded in html.  Still strnage, but I understand.